### PR TITLE
Use GstDeviceMonitor for all the devices!

### PR DESCRIPTION
**Pull request by virtuald**
I know, we want to do a 4.0 release, but this gets rid of a lot of nasty code! And as an added bonus, users can try out the WASAPI sinks if they're suffering from #76

### CODE REVIEW

File: xl/player/gst/sink.py [hunk 1]
```diff
@@ -61,35 +61,65 @@
         "name": "JACK",
         "pipe": "jackaudiosink"
     },
+    "directsoundsink": {
```
Comment (sjohannes) [inline xl/player/gst/sink.py]: I think just `directsound` would be better here, to match the other names.

Also, a WASAPI entry would make sense as well.

**Comment by virtuald** [comment]
Yes, it only works on the automatic sink, as you can't explicitly which sink you want to use when you use the `Gst.DeviceMonitor` stuff (which is why I added the device.api display stuff).

File: xl/player/gst/sink.py (hunk 1 above)
Comment (virtuald) [reply xl/player/gst/sink.py]: The key is stored in settings, so it would break backwards compatibility to change it.

**Comment by virtuald** [comment]
Hm. Looks like Exaile freezes when I start+stop+start a song when using a WASAPI sink (doesn't seem to occur with a directsound sink).

**Comment by virtuald** [comment]
Looks like it's something internal to the WASAPI sink, I get the following error when I set `GST_DEBUG=2`

```
0:00:07.205249000  1456    32bb880 ERROR             wasapisink gstwasapisink.c:687:gst_wasapi_sink_reset:<player-audiosink> IAudioClient::Stop failed (1): Incorrect function.
```
Then when I start playback promptly freezes:

```
0:00:12.997217000  1456    46b43f0 ERROR             wasapisink gstwasapisink.c:613:gst_wasapi_sink_write:<player-audiosink> IAudioClient::Start failed (88890005): AUDCLNT_E_NOT_STOPPED
```
This bug has been reported at https://bugzilla.gnome.org/show_bug.cgi?id=796581 by someone else. I suspect the appropriate fix for it is covered by the patches in https://bugzilla.gnome.org/show_bug.cgi?id=796354

**Pull request merged**